### ISSUE TITLE
TG16/PCE-CD import content and using playlist issue

### ISSUE THREAD

**Issue by retro-wertz** [libretro/RetroArch#4003]
have myself a verified complete dump of Akumajou Dracula X - Chi no Rondo (Japan) (FABT, FACT) based from redump.org dat. using "import content" so i can play using the playlist cannot find the dump with it uncompressed. zipped it now its added to playlist, but when i select it RA crashes. looking at the zipped folder it unzipped only the cue sheet but not the entire cd image. so i extracted the other tracks of image but only zipped the cue sheet, not RA's import content works and adds the game to playlist and you can select it fine...

so to use tg16/pce cd you need to zip the cue sheet and leave the other tracks uncompressed for import and playing from the playlist to work?

of course you can play using load content method but psx works so this should behave the same.

*a quick test can be used but just downloading the verified cue sheet pack from redump.org and do the same procedure above even without the complete cd dump.

**Comment by lasers** [comment]
Confirmed.

**Comment by fr500** [comment]
It's not supported... compressing cue+bins is rather pointless anyway since extracting would take a looooong time and use quite a bit of ram and would need to be performed every single time you load the game

**Comment by lasers** [comment]
What about uncompressed game?  retro-wertz is not able to play uncompressed game in the playlist and I confirmed this. To play this game, you have to use `Load Core` and `Load Content`.

With that same game, the script wrote Track 22.bin instead of cue in `NEC - PC Engine CD - TurboGrafx-CD.lpl`.

Redump is down for days right now so I can't check, but I suspect RetroArch may be hoarding outdated database making it impossible for him to scan uncompressed game.

**Comment by fr500** [comment]
Not sure about databases I'm telling you loading or scanning compressed
bin+cues won't work, that's all

**Comment by retro-wertz** [comment]
@fr500 - the problem here, at least for tg16/pce cd is that you need to compress the cue sheet just for RA to find it when you use IMPORT to make the playlist, its not that you need cue/bin unzipped its that when the cue sheet is uncompressed RA will not find tour rom/cue/cd-image

so its uncompress cd-image, BUT compress the cue sheet in order for RA to find them and play from playlist(assuming we do use the playlist feature and not keep using load content>select file and detect core making playlist useless then)

**Comment by lasers** [comment]
@retro-wertz The list got updated 3 days ago. Update your database and try again.

**Comment by retro-wertz** [comment]
rdb or db is not the problem.its when its when its "cue" based thats causing RA(like pce-cd, sega cd and sega saturn which are cue-sheet based db's) not to import them to playlist unless these cue sheets are zipped. 

like if you unzipped your cue/bin cd-image, IMPORT CONTENT will just do this:

![1](https://cloud.githubusercontent.com/assets/23315382/20869800/60a0fd26-bab5-11e6-95db-3db921b4c6fe.jpg)

its only when you zip the cue-sheet that RA will write this to playlist.

**Comment by lasers** [comment]
How are you zipping the cue sheet? How should it look like? 

I tried (dir: zipping everything inside) -- No go.
I tried (dir: everything unzipped, cue zipped) -- No go.
5 days ago, I did things and was able to add it to playlist. Now I can't and I don't remember what I did. Pinging @bparker06 to see if he can give us answers / work on a solution.

**Comment by retro-wertz** [comment]
![image](https://cloud.githubusercontent.com/assets/23315382/20870105/e9e20eb6-babc-11e6-9227-a478c507524f.png)

**Comment by lasers** [comment]
That picture really helped me out.
I tried everything. All my files originally was this. `...no Rondo (Japan) (FACT)`... No go. 
Changed all files to `... no Rondo (Japan) (FABT, FACT)`... Still no go. Hmm. Wait
Changed all names inside cue too... Still no go. Hmm. Wait.
Zips the cue.  Okay. It got picked up now.
After launching the game, it'll unzip putting cue back in same place.

Sounds about right? Confirmed!

**Comment by retro-wertz** [comment]
yeah, it only puts out CUE sheet. if somebody zips the whole cd, it wont play since it will not extract the rest of the files. and its wierd to have a zipped cue sheet which is only a few KB's and have the rest of the image unzipped in order to play if from playlist(assuming used import content and not manual edit)

**Comment by gcb** [comment]
tried to zip only the cue file. and still no luck. zip without dirs, just the cue file. also left a copy of cue file out of zip, and iso file. everything correctly name in cue file (even case sensitive).

if i load core and load content (cue file), it works fine.

anyway i can have it on the list? if not, can someone share a template playlist? since it doesnt load any cd game, dont even know what to put on the playlist file header

(sorry bad typing, writing from htpc. will clean up later)

**Comment by RobLoach** [comment]
Zipped .img/.cue/.bin files are not really supported. See https://github.com/libretro/RetroArch/issues/4331 for a suggestion to extract the whole archive on .cue files.

When RetroArch files a .bin/.img file, rather than using a CRC check, it attempts to find the serial of the system, and uses that instead. So, while it may match:
https://github.com/libretro/libretro-database/blob/master/metadat/redump/NEC%20-%20PC%20Engine%20CD%20-%20TurboGrafx-CD.dat#L51

It doesn't have any serial information available. If you happen to know a DAT that gives us the serial data for those games, we may be able to get it to work.

**Comment by gcb** [comment]
They are not zipped.

After the first failures, I zipped the cue file because it was mentioned here as a work around. 

The structure is like:

/y.cue (point to "y.iso". Double checked contents from redump.org)
/y.zip (only has y.cue. Not that at first i tried without this file)
/y.iso (correct checksum)

If i select "load content", select the cue file, and then select the engine (it offers me 3 options, PCE, genesis, don't remember the other one, probably psx) then the game loads up fine (and i get the "y" state file in the dir now)

if i select "add content" and select the cue file or the directory, nothing gets added.

**Comment by Nine-H** [comment]
it only indexed my .ccd images and even then it added the wrong files to the playlist and I had to manually change *.img to *.ccd

**Comment by zoltanvb** [comment]
Old ticket, it seems libretro (and maybe also redump) database has changed in the meantime. 

[Current entry](https://github.com/libretro/libretro-database/blob/c2b26123664ee485d91adf6689c35d8966b6fcf5/metadat/redump/NEC%20-%20PC%20Engine%20CD%20-%20TurboGrafx-CD.dat#L83):
```
game (
	name "Akumajou Dracula X - Chi no Rondo (Japan)"
	description "Akumajou Dracula X - Chi no Rondo (Japan)"
	region "Japan"
	serial "KMCD3005"
	rom ( name "Akumajou Dracula X - Chi no Rondo (Japan) (Track 02).bin" size 24399648 crc C681F9DA md5 4CB1CB4858B88122FE4EA413050FD012 sha1 C9B092FB15AF2A7436F425BF29ED5D948F9C9640 serial "KMCD3005" )
)
```
Based on this, the `.bin` file of Track 02 would be detected. Serial numbers are used for several types of disc images, but PCE-CD does not seem to be one of them, [respective database builder script](https://github.com/libretro/libretro-super/blob/9309af6b753bf52f73647e72454ba78a8b2341af/libretro-build-database.sh#L303) refers to crc instead of serial.

**Comment by i30817** [comment]
Several sets are in this situation where the redump or tosec standard uses divided tracks, and RetroArch tries to use a track to "identify" a game. The problem being, a track is by definition not the complete game and kind of useless by itself.

I haven't bothered to read the code, but I would be not surprised if it either use a track as the entry point of the playlist entry (causing bugs) or tried to associate the track to the actual entry point in the database and get that instead (the correct way to solve this but hard), or worse of all, hardcode redump conventions and string transform the found track into the cue (by removing the (track ##) part and uniquifying the resulting spaces or removing them if they're just before extension.

This, and many other problems with using serials is why I stopped using the automatic scanner and learned to love the manual scanner (even with the usability traps it has that can cause people not to get the images if they don't set up the scan exactly right for the dump.)

**Comment by zoltanvb** [comment]
Yep, the "content is one file" approach has its limits, somewhat similar to the RetroPad concept: a necessary simplification, that works very well for the majority of the use cases, but makes it hard to work around when it doesn't. Anyway, no immediate ideas for improvement, but at least the original problem in this issue should not happen nowadays.
